lib/ModuleManager.cpp

```diff
@@ -752,6 +764,12 @@ void QoreModuleManager::loadModuleIntern(ExceptionSink& xsink, const char* name,
 
             mi = loadBinaryModuleFromPath(xsink, name, 0, pgm, reexport);
         }
+        else if (QoreDir::folder_exists(modulePath, xsink)) {
+            qore_offset_t i = modulePath.rfind(QORE_DIR_SEP);
+            // "feature" means pure module name (e.g. "Mime", "CsvUtil" etc.)
+            const char * feature = QoreString(modulePath).extract(++i, &xsink)->c_str();
```

Comment:
@sejvlond @petrycta yes this is a memory leak - `QoreString::extract()` returns a new object that needs to be deleted

Reply:
so if extract returns a new object, shouldn't it be const and so we can call it on `modulePath` directly? or does it also change the object on which it's called?
and should it be freed with `delete` or just `deref`? thx

Reply:
> so if extract returns a new object, shouldn't it be const 
yes it should, but that is not related to this branch or issue

> and so we can call it on modulePath directly? 
it would be best

> or does it also change the object on which it's called?
it should not, it is most likely a `const` error

> and should it be freed with delete or just deref?
with `delete` as it is not a reference-counted class - `QoreStringNode` is a different story

Reply:
> QoreStringNode is a different story
ah, I see
thx